RealmSwift/Handover.swift

```diff
@@ -0,0 +1,181 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+
+#if swift(>=3.0)
+    
+/// A Realm-bound object that can only be passed between threads by exporting for handover
+@objc public protocol ThreadConfined {
```

Comment:
I could add a FIXME to the codebase explaining this so that it can be removed once Swift has the necessary features. (Some tests will not compile with `@objc` removed, so there's no danger in prematurely removing it.)